obs-studio-server/source/main.cpp

```diff
@@ -53,6 +51,49 @@ extern "C" __declspec(dllexport) DWORD NvOptimusEnablement = 1;
 
 #if defined(_WIN32)
 #include "Shlobj.h"
+
+// Checks DisableGPUAsRenderDevice setting
+extern "C" __declspec(dllexport) DWORD NvOptimusEnablement = [] {
+	LPWSTR       roamingPath;
+	std::wstring filePath;
+	std::string  line;
+	std::fstream file;
+	bool         settingValue = false;
+
+	if (FAILED(SHGetKnownFolderPath(FOLDERID_RoamingAppData, 0, nullptr, &roamingPath))) {
+		// Wasn't able to find roaming app data path
+		return 1;
+	} else {
+		filePath.assign(roamingPath);
+		filePath.append(L"\\slobs-client\\basic.ini");
+		CoTaskMemFree(roamingPath);
+	}
+
+	file.open(filePath);
+
+	if (file.is_open()) {
+		while (std::getline(file, line)) {
+			if (line.find("DisableGPUAsRenderDevice", 0) != std::string::npos) {
+				if (line.substr(line.find('=') + 1) == "true") {
+					settingValue = true;
+					file.close();
+					break;
+				}
+			}
+		}
+	} else {
+		// Wasn't able to open config file
+		return 1;
+	}
+
+	if (settingValue) {
+		// Disable high performance graphics rendering
+		return 0;
+	}
+
+	// Enable high performance graphics rendering
+	return 1;
```

Comment:
Can we make it a bit easier to read? We could initialize the `settingValue` to true and only change the it if we successfully find the config file and find the setting in it. Then return the `settingValue` in all cases.

Reply:
I did the changes you asked and ended up changing the name of the setting to be able to return settingValue directly and don't make everything confusing. I believe it is better now.